device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java

```diff
@@ -948,13 +984,15 @@ void setOption_SetAmqpOpenDeviceSessionsTimeout(Object value)
      */
     public void setProxySettings(ProxySettings proxySettings)
     {
+        verifyNotUnregisteredIfMultiplexing();
+
         if (this.deviceIO.isOpen())
         {
             throw new IllegalStateException("Cannot set proxy after connection was already opened");
         }
 
         IotHubClientProtocol protocol = this.deviceIO.getProtocol();
-        if (protocol != HTTPS && protocol != AMQPS_WS && protocol != MQTT_WS)
+        if (protocol != HTTPS && protocol != AMQPS_WS && protocol != MQTT_WS && proxySettings != null)
```

Comment:
good catch, we can probably check to see if it's null at the beginning and throw there. before we check for anything else.